upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -308,6 +308,22 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap, secretStore fi.SecretS
 	dest["IsIPv6Only"] = tf.IsIPv6Only
 	dest["UseServiceAccountExternalPermissions"] = tf.UseServiceAccountExternalPermissions
 
+	if cluster.Spec.ClusterAutoscaler != nil {
+		dest["ClusterAutoscalerPriorities"] = func() map[string][]string {
+			priorities := make(map[string][]string)
+			for name, spec := range tf.GetNodeInstanceGroups() {
+
+				if spec.Autoscale != nil {
+					priorities[fmt.Sprint(spec.AutoscalePriority)] = append(priorities[fmt.Sprint(spec.AutoscalePriority)], fmt.Sprintf("%s.%s", name, tf.ClusterName()))
+				}
+			}
+			return priorities
+		}
+		dest["GetClusterAutoscalerExpander"] = func() string {
```

Comment:
Why is this function required? Can't you use the value directly?

Reply:
Without this function `eq` failed on template rendering with error `error calling eq: invalid type for comparison`

It seems that dereferencing pointers does not work with `eq`

Reply:
Hm. Expander shouldn't even be a pointer as `""` (empty string)  isn't a valid value and we always set this one.

Let me do a quick PR for that